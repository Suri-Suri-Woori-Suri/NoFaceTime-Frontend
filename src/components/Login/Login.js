import React from 'react';


const Login = () => {
  return <div>hello this is login</div>;
};

export default Login;
