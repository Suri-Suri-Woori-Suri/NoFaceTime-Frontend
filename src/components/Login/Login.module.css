.LoginWrap {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: calc(100vh - 150px);
  /* background-color: #e5eaf5; */
  background-color: whitesmoke;
}

.Login {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: calc(max(300px, 20vw));
  height: calc(max(500px, 50vh));
  background-color: whitesmoke;
  border-radius: 3%;
  box-shadow: 0px 10px 25px grey;
  font-family: "Oswald";
}

.GoogleLoginButton {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80%;
  min-width: 100px;
  height: 40px;
  padding: 0px 10px 0px 10px;
  background: #274ba0;
  color: #ffffff;
  border: 1px solid #274ba0;
  border-radius: 20px 20px 20px 20px;
  font-family: "Open Sans";
  font-weight: 700;
  font-size: 1rem;
  text-align: center;
  box-shadow: 0px 5px 10px grey;
}

.GoogleLoginButton:hover {
  transform: translateY(2px);
  box-shadow: 0px 4px 5px grey;
}
