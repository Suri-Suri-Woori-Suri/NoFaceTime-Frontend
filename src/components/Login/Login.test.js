import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen } from '@testing-library/react';

import Login from './Login';

describe('<Login />', () => {
  const mockUpdateUserData = jest.fn();
  const wrapper = mount(<Login updateUserData={mockUpdateUserData} />);

  beforeEach(() => {
    render(<Login updateUserData={mockUpdateUserData} />);
  });

  afterEach(cleanup);

  it(`should be contain 'Login' text`, () => {
    expect(screen.getByText('Login')).toBeInTheDocument(true);
  });

  it(`should be contain 'Login' button`, () => {
    expect(screen.getByText('Google Login')).toBeInTheDocument(true);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
