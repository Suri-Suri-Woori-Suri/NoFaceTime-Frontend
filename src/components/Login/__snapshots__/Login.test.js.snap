// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<Login
  updateUserData={[MockFunction]}
>
  <div
    className="LoginWrap"
  >
    <div
      className="Login"
    >
      <h1>
        Login
      </h1>
      <button
        className="GoogleLoginButton"
        onClick={[Function]}
      >
        Google Login
      </button>
    </div>
  </div>
</Login>
`;
